src/Containers/HorizontalBarChart/HorizontalBarChart.tsx

```diff
@@ -0,0 +1,168 @@
+import React,{useCallback, useLayoutEffect, useRef,useState,useMemo} from 'react'
+import {BarChart,XAxis,YAxis,Bar,Text,Cell} from 'recharts'
+import styled from 'styled-components'
+import Theme from '@Themes/ThemeTemplate'
+import {MainTheme} from '@Themes'
+import {Mixins} from '@Utils'
+import {Paragraph} from '@Text'
+
+const FLEX_EQUAL_SPACE=1
+const SUMMARY_BACKGROUND_COLOR='#d3d3d3'
+const BAR_LABEL_LEFT_MARGIN=5
+const SUMMARY_CONTAINER_MARGIN=14
+const IS_TOTAL_COLOR='#d3d3d3'
+const GREEN_COLOR_MARGIN=5
+const ORANGE_COLOR_MARGIN=15
+const BAR_SIZE=20
+
+interface IDataItem{
+    label:string;
+    value:number;
+    isTotal:boolean;
+}
+
+interface IChartOptionalProperties{
+    margin?:object;
+}
+
+export interface IHorizontalBarChartProps extends IChartOptionalProperties{
+    data:IDataItem[];
+    header:string;
+    summaryHeader:string;
+    summaryDescription:string;
+    unit:string;
+    width?:number;
+}
+
+export const HorizontalBarChart:React.FC<IHorizontalBarChartProps>=({data,header,summaryHeader,summaryDescription,width,margin,unit}):React.ReactElement=>{
+    const [chartWidth,setChartWidth]=useState<number>()
+    const [chartHeight,setChartHeight]=useState<number>()
+
+    // used to compute width and height for the chart based on width and height of summary container
+    const summaryContainerRef=useRef<HTMLDivElement>(null)
+
+    /**
+     * gets an object with not undefined properties for the chart
+     * @returns {IChartOptionalProperties} an object with the not undefined optional properties set to its value
+     */
+    const getChartOptionalProperties=()=>{
+        const chartOptionalProperties:IChartOptionalProperties={}
+        if(margin)
+            chartOptionalProperties.margin=margin
+        return chartOptionalProperties
+    }
+
+    /**
+     * compute dynamically the width and height of the chart
+     */
+    useLayoutEffect(()=>{
```

Comment:
Why?

Reply:
I was trying to mimic the design you showed me

Reply:
I mean why do you need to dyamically set the width?

Reply:
Won't Flexbox do it?

Reply:
Flex with divs